toolchain/check/decl_name_stack.h

```diff
@@ -153,13 +153,13 @@ class DeclNameStack {
       // expressions. `None` indicates resolution failed.
       SemIR::InstId resolved_inst_id;
 
-      // The ID of an unresolved identifier.
-      SemIR::NameId unresolved_name_id = SemIR::NameId::None;
+      // When `state` is `Poisoned` (name is unresolved due to name poisoning),
+      // the poisoning location.
+      SemIR::LocId poisoning_loc_id = SemIR::LocId::None;
     };
 
-    // When `state` is `Poisoned` (name is unresolved due to name poisoning),
-    // the poisoning location.
-    SemIR::LocId poisoning_loc_id = SemIR::LocId::None;
+    // The ID of an identifier.
+    SemIR::NameId name_id = SemIR::NameId::None;
```

Comment:
I'm realizing you're changing this more than just the diagnostic. Can you please update the PR and explain why you're changing semantics here? Looking at the PR, it's not clear to me why this should be changing.

Reply:
I've rephrased the PR description text to make it clearer.

Reply:
I can split this PR to two if you prefer.

Reply:
No, I wasn't asking for the PR to be split, just for you to document your changes. Thanks!